sortable directive type

yonathan06 opened the pull request:
Sortable directive is not added to the JSX directive typing, leading to a typescript error

example: https://stackblitz.com/edit/solid-vite-rtvzss?file=src%2Findex.tsx

martinpengellyphillips commented:
Thanks for sharing this. 

I'm not super familiar yet with typescript, but I believe this declaration _has_ to be made in the consumer code because you can name the variable anything you like. If we added this declaration to solid-dnd it would still error if your example looked like:

```js
const Item = (props) => {
  const foo = createSortable(props.todo.id);
  return <li use:foo>{props.todo.text}</li>;
};
```

yonathan06 commented:
Good point, added a commit to have any key string, tried to see if there is a way in TypeScript to infer the user var name, but didn't find one. Maybe it the return value will be an object, something like this: `  const { sortable } = createSortable(...);` then TypeScript could use the key name. I added a boolean because the directive is used as a boolean prop (equal to true)

I know it is a little ugly patch, but I think having the developer to add a `declare` type annotation is bad DX, as not everyone are familiar with advanced typings. Will happy to hear what you think on this small change

martinpengellyphillips commented:
This would match any directive, even those unrelated to solid-dnd, so I don't think it makes sense. It could cause poorer dx from confusion of types.

This might be something to raise more generally in the SolidJS community I think as it's not a problem specific to solid-dnd. It affects any solid library that exposes directives.

yonathan06 commented:
You are right, it is worth mentioning it on solid's repo. added a new change suggesting to return the sortable object as a prop, and have the key `sortable` in the type declaration (other than a general key). I know this is a change of behavior and can break things, so I see this as a suggestion until there will be a general solution from solid.
